test/gtest/ucp/test_ucp_proto_select.cc

```diff
@@ -0,0 +1,447 @@
+            g_component_map.emplace(component,
+                                    std::unique_ptr<mock_component>(mock_comp));
+        }
+
+        return *mock_comp;
+    }
+
+    static void cleanup()
+    {
+        g_component_map.clear();
+        mock_t::cleanup();
+    }
+
+    uct_component_t * operator->() const
+    {
+        return m_component;
+    }
+
+    void set_mock_perf_attr(const std::string &iface_name, perf_attr_func_t cb)
+    {
+        m_perf_attrs_funcs[iface_name] = cb;
+    }
+
+    void set_mock_iface_attr(const std::string &iface_name,
+                             iface_attr_func_t cb)
+    {
+        m_iface_attrs_funcs[iface_name] = cb;
+    }
+
+    static void set_mock_memory_distance(const ucs_sys_dev_distance_t &distance)
+    {
+        g_memory_distance = distance;
+
+        ucs_sys_topo_provider_t *provider;
+        ucs_list_for_each(provider, &ucs_sys_topo_providers_list, list) {
+            mock_t::setup(&provider->ops.get_memory_distance,
+                (ucs_topo_get_memory_distance_func_t)(
+                [](ucs_sys_device_t device, ucs_sys_dev_distance_t *distance) {
+                    *distance = g_memory_distance;
+                }));
+        }
+    }
+
+private:
+    mock_component(uct_component_h component) :
+        m_component(component),
+        m_flags_orig(component->flags)
+    {
+        uct_tl_t *tl;
+        ucs_list_for_each(tl, &component->tl_list, list) {
+            mock_t::setup(&tl->iface_open, iface_open_mock);
+        }
+    }
+
+    static uct_component_h find_uct_component(const std::string &name)
+    {
+        uct_component_h component;
+        ucs_list_for_each(component, &uct_components_list, list) {
+            if (name == component->name) {
+                return component;
+            }
+        }
+
+        return nullptr;
+    }
+
+    static mock_component * find(uct_component_h component)
+    {
+        auto it = g_component_map.find(component);
+        return (it == g_component_map.end()) ? nullptr : it->second.get();
+    }
+
+    uct_tl_t * find_tl(const uct_iface_params_t *params) const
+    {
+        uct_tl_t *tl;
+        ucs_list_for_each(tl, &m_component->tl_list, list) {
+            if (0 == strcmp(tl->name, params->mode.device.tl_name)) {
+                return tl;
+            }
+        }
+
+        return nullptr;
+    }
+
+    static ucs_status_t iface_open_mock(uct_md_h md, uct_worker_h worker,
+                                        const uct_iface_params_t *params,
+                                        const uct_iface_config_t *config,
+                                        uct_iface_h *iface_p)
+    {
+        mock_component *self = find(md->component);
+        uct_tl_t *tl         = self->find_tl(params);
+        ucs_status_t status  = mock_t::actual_call(&tl->iface_open, md, worker,
+                                                   params, config, iface_p);
+        if (status == UCS_OK) {
```

Comment:
Minor: invert branches in all these places. Better to have:
```
if (status != UCS_OK) {
    return status;
}

...
...
return UCS_OK;
```

Reply:
ok